libusb/core.c

```diff
@@ -2308,7 +2308,7 @@ int API_EXPORTED libusb_init(libusb_context **ctx)
 #endif
 
 	/* default context should be initialized before calling usbi_dbg */
-	if (!ctx) {
+	if (!usbi_default_context) {
```

Comment:
Maybe this can fix performance, but this breaks a logic: if `libusb_init(ctx)` is called with `ctx != null`, then `usbi_default_context` will be the same as `ctx`. And after `libusb_exit(ctx)` is called, `usbi_default_context` will point to the garbage.

Reply:
Good point. Thanks for pointing this out.